Simplenote/NoteContentHelper.swift

```diff
@@ -0,0 +1,82 @@
+import Foundation
+
+// MARK: - NoteContentHelper
+//
+struct NoteContentHelper {
+
+    private init() {
+
+    }
+
+    /// Returns structure of the content: range of title and body
+    ///
+    /// - Parameters:
+    ///     - content: note content
+    ///
+    static func structure(of content: String?) -> (title: NSRange, body: NSRange) {
+        let titleRange = trimmedTextRange(in: content, startingFrom: 0, endAtNewline: true)
+        let bodyRange: NSRange
+        if titleRange.isNotFound {
+            bodyRange = .notFound
+        } else {
+            bodyRange = trimmedTextRange(in: content, startingFrom: NSMaxRange(titleRange), endAtNewline: false)
+        }
+
+        return (
+            title: titleRange,
+            body: bodyRange
+        )
+    }
+
+    private static func trimmedTextRange(in content: String?, startingFrom startLocation: Int, endAtNewline: Bool) -> NSRange {
+        guard let content = content else {
+            return .notFound
+        }
+        let fullRange = content.nsString.fullRange
+
+        // Look for the first character ignoring whitespaces and newlines
+        let firstCharacterSearchRange = NSRange(location: startLocation,
+                                                length: fullRange.length - startLocation)
+        let firstCharacterRange = content.nsString.rangeOfCharacter(from: CharacterSet.whitespacesAndNewlines.inverted,
+                                                                    options: [],
+                                                                    range: firstCharacterSearchRange)
+        guard !firstCharacterRange.isNotFound else {
+            return .notFound
+        }
```

Comment:
Q: Is this trimming improving performance?

Reply:
Probably not, but I also think it's not affecting performance much. And if we do it here, we wouldn't need to do it when showing the string in a label.